packages/next/shared/lib/flush-effects.tsx

```diff
@@ -1,8 +1,8 @@
-import React, { createContext, useContext } from 'react'
+import React, { useContext } from 'react'
 
 export type FlushEffectsHook = (callbacks: () => React.ReactNode) => void
 
-export const FlushEffectsContext = createContext<FlushEffectsHook | null>(
+export const FlushEffectsContext = React.createContext<FlushEffectsHook | null>(
```

Comment:
will this also effect other shared components that only imported by client components on user land? this file shouldn't be treated as server component

Reply:
@huozhi I explained in the PR description, this file isn't **used** as a server component, however but it's imported as a dependency in the server graph, because we don't have perfect tree-shaking.